sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -640,13 +679,20 @@ def _resolve_dependencies_for_pipeline_component_jobs(
             component,
```

Comment:
What's the benefit for divide nodes into layers compared to in-order traversal?

Reply:
It's a preparation for concurrent node resolution. Nodes in the same layer can be resolved concurrently if lower layers have already been resolved.

Reply:
To further explain, we'd better not to resolve a pipeline component and one of its jobs concurrently, as hash calculation of the pipeline component will be impacted by the resolution of its children.